DataDog/datadog-operator
# Fix initContainer overrides

### What does this PR do?

Fix container overrides for init containers: 

* name
* resources
* volumeMounts
* env vars
* securityContext (not including secomp profile)

Some container overrides were omitted because they are not supported in init containers (livenessProbe and readinessProbe), may cause regression in functionality (args, command), or would have no effect (DD healthPort). 

This PR also fixes override container unit tests as well as adds new unit tests for init container overrides. 

### Motivation

Bug fix and customer support case

### Additional Notes

Anything else we should know when reviewing?

### Minimum Agent Versions

Are there minimum versions of the Datadog Agent and/or Cluster Agent required?

* Agent: N/A
* Cluster Agent: N/A

### Describe your test plan
Test in kind cluster and non-kind cluster:
1. Deploy operator
2. Apply `DatadogAgent` manifest overriding init container configs
3. Check that overrides are applied: `kubectl get pod -oyaml <agent_pod_name>`
 
```
apiVersion: datadoghq.com/v2alpha1
kind: DatadogAgent
metadata:
  name: datadog
spec:
  global:
    clusterName: kind-kind  # needed for kind
    kubelet:
      tlsVerify: false  # needed for kind
    credentials:
      apiSecret:
        secretName: datadog-secret
        keyName: api-key
      appSecret:
        secretName: datadog-secret
        keyName: app-key
#  features:  # can't test system-probe in kind
#    npm:
#      enabled: true
  override:
    nodeAgent:
      containers:
        agent:
          env:
            - name: DD_LOG_LEVEL
              value: debug
        init-config:
          name: new-init-config-name
          env:
            - name: INIT_ENV_VAR
              value: SOME_VALUE
          resources:
            limits:
              cpu: "3"
              memory: "2Gi"
            requests:
              cpu: "2"
              memory: "1Gi"
        init-volume:
          resources:
            limits:
              cpu: "3"
              memory: "2Gi"
            requests:
              cpu: "2"
              memory: "1Gi"
          volumeMounts:
          - name: custom-volume
            mountPath: /custom-volume
          securityContext:
            runAsUser: 0
#        seccomp-setup:  # can't test system-probe in kind
#          securityContext:
#            runAsUser: 0
      volumes:
        - emptyDir: {}
          name: custom-volume
```

### Checklist
